### PR TITLE
SAP system details page tests

### PR DESCRIPTION
**Pull request by arbulu89**
This PR implements the e2e tests for the SAP system details page.

I have added a bit more of testing than the strictly required. For example, I have started playing with state changes, to see that the status badge is properly change. For that, I have added more scenario. They are sent by photofinish. Actually I thought to implement a standalone method to send specific jsons, but well, this way it works too hehe

In essence, the HANA database details page is exactly the same, but I prefer to open a new PR with that, to make the review a bit easier.

### CODE REVIEW

File: test/e2e/cypress/fixtures/sap-system-details/sap_system_details.feature [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        Given I navigate to a specific SAP system ('/sapsystems/a1e80e3e152a903662f7882fb3f8a851')
+        Then 4 instances are displayed
+        And the data of each instance is correct
+        And the status of each instance is GREEN
+
+    Scenario: SAP system instances status change event is received
+        Given I navigate to a specific SAP system ('/sapsystems/a1e80e3e152a903662f7882fb3f8a851')
+        When a new SAP system event for this system with the 1st instance with a GRAY status is received
+        And the page is refreshed
+        And the status of the 1st instance is GRAY
```
Comment (nelsonkopliku) [inline test/e2e/cypress/fixtures/sap-system-details/sap_system_details.feature:30]: Is this an intended `And` or is it a typo and it is actually a `Then`?

**Review by nelsonkopliku — COMMENTED** [review]
Wow man, this is really nice!

I do really appreciate the effort in writing `feature` files, you're doing a great job, thanks!
Helps a lot in understanding the test.

Waiting for the build to succeed :wink:

File: test/e2e/cypress/fixtures/sap-system-details/sap_system_details.feature (hunk 1 above)
Comment (arbulu89) [reply test/e2e/cypress/fixtures/sap-system-details/sap_system_details.feature:30]: Good catch, The last one should be a `Then`

**Comment by arbulu89** [comment]
More things:
- I have added an `artifact-uploads` steps if the e22 tests fail. It comes handy to check the video
- Update `sap_system` service to return the instances ordered by instance_number. This was the reason why the tests were failing, Now, they are always coming in the same order, and ordering by instance number makes sense in my opinion

**Review by nelsonkopliku — APPROVED** [review]
Aha! Ordering! We all had the same issue :smile:

**Pull request merged**